test/hotspot/jtreg/compiler/intrinsics/unsafe/ByteBufferTest.java

```diff
@@ -217,7 +217,9 @@ void ck(long x, long y) {
 
     void ck(double x, double y) {
         if (x == x && y == y && x != y) {
```

Comment:
What are the `x == x` and `y == y` checks good for?

Reply:
I removed it. @PaulSandoz agreed that this method was very suspect. I should try removing it. There is an off chance it could create some NaN issues on odd platforms.